WalletWasabi/Crypto/ZeroKnowledge/ZkVerifier.cs

```diff
@@ -0,0 +1,25 @@
+using NBitcoin.Secp256k1;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using WalletWasabi.Helpers;
+
+namespace WalletWasabi.Crypto.ZeroKnowledge
+{
+	public static class ZkVerifier
+	{
+		public static bool Verify(ZkProof proof)
+		{
+			Guard.NotNull(nameof(proof), proof);
+
+			var publicPoint = proof.PublicPoint;
+			var randomPoint = proof.RandomPoint;
+			var challenge = ZkChallenge.Build(publicPoint, randomPoint);
+
+			var left = (publicPoint * challenge + randomPoint).ToGroupElement();
+			var right = (EC.G * proof.Response).ToGroupElement();
+			return (left.IsInfinity && right.IsInfinity) || (left.x == right.x && left.y == right.y);
```

Comment:
This wrong. 

```c#
var fex = new FE(123456);
var ge1 = new GE(fex, fex, false);
var ge2 = new GE(fex, fex, true);

/ ge1 and ge2 are no equal but this will pass
Assert.True((ge1.IsInfinity && ge2.IsInfinity) || (ge1.x == ge2.x && ge1.y == ge2.y))

// this will not pass
Assert.Equal(ge1, ge2);
```

The real equality is  achieved by `==` because it means that is exactly (byte to byte) equal. You can achieve something similar by:

```c#
return left.x == right.x && left.y == right.y && left.IsInfinity == right.IsInfinity;
```
But IMO this makes no sense.